utils/src/main/scala/za/co/absa/enceladus/utils/general/ConfigReader.scala

```diff
@@ -31,4 +38,41 @@ object ConfigReader {
   def readStringConfig(path: String, default: String): String = {
     readStringConfigIfExist(path).getOrElse(default)
   }
+
+  /**
+   * Given a configuration returns a new configuration which has all sensitive keys redacted.
+   *
+   * @param keysToRedact  A set of keys to be redacted.
+   */
+  def getRedactedConfig(keysToRedact: Set[String]): Config = {
+    def addToConfigIfIn()(accumulatedConfig: Config, key: String): Config = {
+      if (config.hasPath(key)) {
+        accumulatedConfig.withValue(key, ConfigValueFactory.fromAnyRef(redactedReplacement))
+      } else {
+        accumulatedConfig
+      }
+    }
+
+    val redactingConfig = keysToRedact.foldLeft(ConfigFactory.empty)(addToConfigIfIn())
+
+    redactingConfig.withFallback(config)
+  }
+
+  /**
+   * Logs the effective configuration while redacting sensitive keys.
+   *
+   * @param keysToRedact A set of keys for which values shouldn't be logged.
+   */
+  def logEffectiveConfig(keysToRedact: Set[String] = Set()): Unit = {
+    val redactedConfig = getRedactedConfig(keysToRedact)
+
+    val renderOptions = ConfigRenderOptions.defaults()
+      .setComments(false)
+      .setOriginComments(false)
+      .setJson(false)
+
```

Comment:
Not quite sure about this decision. Maybe `ConfigRenderOptions.concise()` is better since it is JSON and it can be easily formatted when investigating a log.

Reply:
As stated on the CQC, HOCON-dotted notation would be advantageous for searching in the output.